specification/ai/OpenAI.Assistants/runs/models.tsp

```diff
@@ -0,0 +1,187 @@
+
+using TypeSpec.Versioning;
+
+@doc("Data representing a single evaluation run of an assistant thread.")
+@added(ServiceApiVersions.v2023_11_06_beta)
+model ThreadRun {
+  @doc("The identifier, which can be referenced in API endpoints.")
+  id: string;
+
+  @doc("The object type, which is always 'thread.run'.")
+  object: "thread.run";
+
+  @projectedName("json", "thread_id")
+  @doc("The ID of the thread associated with this run.")
+  threadId: string;
+
+  @projectedName("json", "assistant_id")
+  @doc("The ID of the assistant associated with the thread this run was performed against.")
+  assistantId: string;
+
+  @doc("The status of the assistant thread run.")
+  status: RunStatus;
+
+  // Internal note: API reference describes this as nullable and the spec explicitly defines it as required and
+  //                nullable, but actual test traffic confirms that runs without tool calls do not present
+  //                required_action on their responses. We split the difference to be optional and nullable.
+  //                https://platform.openai.com/docs/api-reference/runs/object#runs/object-expires_at
+  //                https://github.com/openai/openai-openapi/blob/master/openapi.yaml#L8245
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "explicitly required+nullable in mirrored API"
+  @projectedName("json", "required_action")
+  @doc("The details of the action required for the assistant thread run to continue.")
+  requiredAction?: RequiredAction | null;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "explicitly required+nullable in mirrored API"
+  @projectedName("json", "last_error")
+  @doc("The last error, if any, encountered by this assistant thread run.")
+  lastError: RunError | null;
+
+  @doc("The ID of the model to use.")
+  `model`: string;
+
+  @doc("The overridden system instructions used for this assistant thread run.")
+  instructions: string;
+
+  @doc("The overridden enabled tools used for this assistant thread run.")
+  tools: ToolDefinition[];
+
+  @projectedName("json", "file_ids")
+  @doc("A list of attached file IDs, ordered by creation date in ascending order.")
+  fileIds: string[];
+
+  @projectedName("json", "created_at")
+  @encode(DateTimeKnownEncoding.unixTimestamp, int32)
+  @doc("The Unix timestamp, in seconds, representing when this object was created.")
+  createdAt: utcDateTime;
+
+  // Internal note: API reference describes this as non-nullable, but retrieveThreadRun gets a null expiresAt
+  //                for completed runs.
+  //                https://platform.openai.com/docs/api-reference/runs/object#runs/object-expires_at
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "explicitly required+nullable in mirrored API"
+  @projectedName("json", "expires_at")
+  @encode(DateTimeKnownEncoding.unixTimestamp, int32)
+  @doc("The Unix timestamp, in seconds, representing when this item expires.")
+  expiresAt: utcDateTime | null;
```

Comment:
This field is not nullable. But I only read the comment above after typing this :D

Reply:
Yeah, this one was rough to diagnose at first. We should keep checking up on this but there's currently a clear mismatch for completed runs.